crypto/rand/prov_seed.c

```diff
@@ -44,31 +45,13 @@ size_t ossl_rand_get_user_entropy(OSSL_LIB_CTX *ctx,
                                   unsigned char **pout, int entropy,
                                   size_t min_len, size_t max_len)
 {
-    unsigned char *buf;
     EVP_RAND_CTX *rng = ossl_rand_get0_seed_noncreating(ctx);
-    size_t ret;
 
-    if (rng == NULL)
+    if (rng != NULL && evp_rand_can_seed(rng))
+        return evp_rand_get_seed(rng, pout, entropy, min_len, max_len,
+                                 0, NULL, 0);
+    else
         return ossl_rand_get_entropy(ctx, pout, entropy, min_len, max_len);
-
-    /* Determine how many bytes to generate */
```

Comment:
Should we retain this code path for the case where `rng != NULL && !evp_rand_can_seed(rng)` ?  Existing seed sources are not going to support the new callbacks and will devolve to the current behaviour of gathering entropy themselves.  Fixing that was primary intent behind the earlier change.

Reply:
> Should we retain this code path for the case where `rng != NULL && !evp_rand_can_seed(rng)` ? Existing seed sources are not going to support the new callbacks and will devolve to the current behaviour of gathering entropy themselves.

I thought about that option as well and discarded it. In view of what I wrote in https://github.com/openssl/openssl/pull/22394#issuecomment-1774773581 it is not necessary to 'fix' the case where the seed source doesn't implement the get_entropy hooks. It just makes things more convoluted.

>  Fixing that was primary intent behind the earlier change.

The intention was to fix the VMS low entropy seeding problem, which this pull request achieves as well.